ui/jetbrains/src/main/java/com/neueda/jetbrains/plugin/graphdb/jetbrains/ui/datasource/tree/DataSourcesTreeMouseAdapter.java

```diff
@@ -0,0 +1,67 @@
+package com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.tree;
+
+import com.intellij.ide.DataManager;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.ui.popup.JBPopupFactory;
+import com.intellij.openapi.ui.popup.ListPopup;
+import com.intellij.ui.treeStructure.Tree;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.component.datasource.state.DataSourceApi;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.actions.DataSourceActionGroup;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.metadata.actions.MetadataActionGroup;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.metadata.dto.ContextMenu;
+
+import javax.swing.*;
+import javax.swing.tree.TreePath;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+public class DataSourcesTreeMouseAdapter extends MouseAdapter {
+
+    private ContextMenuService contextMenuService = new ContextMenuService();
+
+    @Override
+    public void mouseClicked(MouseEvent e) {
+        if (SwingUtilities.isRightMouseButton(e)) {
+            Tree tree = (Tree) e.getComponent();
+            TreePath pathForLocation = tree.getPathForLocation(e.getX(), e.getY());
+
+            contextMenuService.getContextMenu(pathForLocation)
+                    .ifPresent(dto -> popup(dto, tree));
+        }
+    }
+
+    private void popup(ContextMenu dto, Tree tree) {
+        DataContext dataContext = DataManager.getInstance().getDataContext(tree);
+        if (dto.getMetadataType() == Neo4jTreeNodeType.LABEL
```

Comment:
I think that `switch` will be better here instead of ifs (because we are using enums)